### ISSUE TITLE
[nvmf/vfio-user] incorrect handling of I/O queue creation error

### ISSUE THREAD

**Issue by tmakatos** [spdk/spdk#2784]
In `handle_queue_connect_rsp`, if the operation failed (the `spdk_nvme_cpl_is_error(&req->req.rsp->nvme_cpl)`) is taken, for I/O queue creation errors we don't post a completion with an SC/SCT set accordingly; instead we silently fail. This is against the spec.

**Comment by changpe1** [comment]
> In `handle_queue_connect_rsp`, if the operation failed (the `spdk_nvme_cpl_is_error(&req->req.rsp->nvme_cpl)`) is taken, for I/O queue creation errors we don't post a completion with SC/SCT set accordingly; instead we silently fail. Isn't this out of spec?

Yes, existing code is fine to process ADMIN QP connection, but not correct for CREATE IO SQ command, would you like to submit a fix patch?

**Comment by wanghailiangx** [comment]
@tmakatos Do you have an update? If there is a patch, please post it.

**Comment by tmakatos** [comment]
Not yet, I'm not planning on working on this for now.
